src/include/ipc/probe.h

```diff
@@ -94,7 +94,7 @@ struct probe_data_packet {
 	uint32_t checksum;		/**< CRC32 of header and payload */
 	uint32_t data_size_bytes;	/**< Size of following audio data */
 	uint32_t data[];		/**< Audio data extracted from buffer */
-} __attribute__((packed));
+};
```

Comment:
can you clarify why it's unnecessary? It's used in all other structures defined in this file.

Reply:
This struct is full of uint32_t which means it will work exactly the same way with and without attribute packed.
So there is no reason to generate false warnings because of forced alignment 1

Reply:
@abonislawski That's not really guaranteed. I don't personally like use of packed structs very much either to implement binary interfaces, but it is faster, so most projects just use them -- and SOF uses them a looot. But, but, without packed it is not guaranteed the struct layout is what you expect, even if it just contains "uint32_t" members. Compiler is free to e.g. align all members to 64bit boundaries if it chooses to. So if this struct is used to cast from/to binary data, then it should be packed.